### ISSUE TITLE
pyright incorrectly allows & parses `await` in lambdas

### ISSUE THREAD

**Issue by jakkdl** [microsoft/pyright#9406]
**Describe the bug**
`await` statements are never allowed in lambda's, but pyright sees no problem with them and treats them as if they're fully valid and will work. Trying to run a file with await in lambda will give a `SyntaxError`, and mypy also gives an error.

**Code or Screenshots**
```python
from typing_extensions import assert_type

async def foo() -> int:
    x = lambda: await foo()
    assert_type(x(), int)
    return 5

```

```sh
$ pyright foo.py
0 errors, 0 warnings, 0 informations 
```

**VS Code extension or command-line**
command line, version 1.1.388

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.389
